backend/pystac_io_methods.py

```diff
@@ -0,0 +1,28 @@
+from typing import IO, Any, Union
+from urllib.parse import urlparse
+
+import boto3
+from pystac import STAC_IO  # type: ignore[import]
+
+
+def read_method(uri: str) -> Any:
+    parsed = urlparse(uri)
+    if parsed.scheme == "s3":
+        bucket = parsed.netloc
+        key = parsed.path[1:]
+        s3_resource = boto3.resource("s3")
+        obj = s3_resource.Object(bucket, key)
+        return obj.get()["Body"].read().decode("utf-8")
```

Comment:
We've used `S3_CLIENT.get_object(Bucket=bucket, Key=key)["Body"]` to get the equivalent elsewhere. We should probably use the same in both places, but I don't know whether there is any appreciable difference between these options.